DOCS-7128 added new content (#240)
mulesoft/docs-api-manager: modules/ROOT/pages/https-reference.adoc

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: configure, HTTPS endpoint

> You automatically deploy an API proxy using an HTTP listener, inbound HTTP endpoint, with minimal setup. You can use this approach if you do not foresee any security concerns in your Anypoint Platform API setup. Otherwise, you must configure your API proxy with an HTTPs listener (inbound HTTPS endpoint), instead of an HTTP listener.

Reviewer: "HTTP listener, inbound HTTP endpoint," -> "HTTP listener(inbound HTTP endpoint)"

Author: Done! Thanks!

> You automatically deploy an API proxy using an HTTP listener, inbound HTTP endpoint, with minimal setup. You can use this approach if you do not foresee any security concerns in your Anypoint Platform API setup. Otherwise, you must configure your API proxy with an HTTPs listener (inbound HTTPS endpoint), instead of an HTTP listener.

Reviewer: You "can". You not always do that. -> "You can automatically deploy an API proxy"

Author: Done! Thanks!

> You can automatically deploy an API proxy using an HTTP listener (inbound HTTP endpoint) with minimal setup. You can use this approach if you do not foresee any security concerns in your Anypoint Platform API setup. Otherwise, you must configure your API proxy with an HTTPs listener (inbound HTTPS endpoint), instead of an HTTP listener.

Reviewer 2: ```suggestion
You can automatically deploy an API proxy using an HTTP Listener (inbound HTTP endpoint) with minimal setup. Use this approach if you are certain that your Anypoint Platform API setup is secure. Otherwise, configure your API proxy with an HTTPS Listener (inbound HTTPS endpoint) instead of an HTTP Listener.
```

Reviewer 2: GLOBALLY change HTTPs to HTTPS. I will stop marking this, unless it is in the course of making another edit.

Author: Done! Thanks!

> MuleSoft requires that you provide your own certificate when configuring your HTTPs setup. The certificate you use can be a self-signed certificate to be used for demo purposes, custom TLS certificate with validity within your organization, or a trusted Certificate Authority (CA) certificate.

Reviewer 2: Consider this reformatting to emphasize the list:

```suggestion
MuleSoft requires that you provide your own certificate when configuring your HTTPS setup. Use one of the following types:
* A self-signed certificate used for demo purposes
* A custom TLS certificate with validity within your organization
* A trusted Certificate Authority (CA) certificate
```

Author: Done! Thanks!

> If your backend uses HTTPs with a trusted CA certificate, you do not need to perform any special action to communicate with it from the API proxy. If your backend uses HTTPs with a self-signed certificate or custom TLS certificate, you must configure the API proxy requester with the corresponding certificate.

Reviewer 2: ```suggestion
If your backend uses HTTPs with a trusted CA certificate, you do not need to perform any special action to communicate with it from the API proxy. 
+
+ If your backend uses HTTPs with a self-signed certificate or custom TLS certificate, you must configure the API proxy requester with the corresponding certificate:
```

Author: Leaving it as is.

Author: Actually changed. Done! Thanks!

> Depending on your business needs, you can use any of the following recommended ways to configure your API Proxy to use HTTPs:

Reviewer: "Depending on your business needs, you can use any of the following recommended ways to configure your API Proxy to use HTTPs:" -> "Depending on your business use case, you can use any of the following three recommended ways to configure your API Proxy to use a certificate when setting up an HTTPS listener:"

Author: Done! Thanks!

> Depending on your business use case, you can use any of the following recommended ways to configure your API proxy to use a certificate when setting up an HTTPS listener:

Reviewer 2: I think we can omit this in favor of the previous sentence, or tweaking the previous sentence a bit.
```suggestion

```

Author: Done! Thanks!

> * xref:building-https-proxy.adoc[HTTPs configured through the UI by leveraging Secret Manager]:

Reviewer: "HTTPs configured through the UI by leveraging Secret Manager" -> "Certificate configured through the UI by leveraging Secret Manager"

Author: Done! Thanks!

> * xref:building-https-proxy.adoc[Certificate configured through the UI by leveraging Secret Manager]:

Reviewer 2: ```suggestion
* xref:building-https-proxy.adoc[Configure a certificate through the UI by leveraging the secrets manager.
```

Author: No Secrets Manager is the name of the feature or product. Done! Thanks!

+ 

> The certificate is uploaded to a secure vault within API Platform, and then made available when configuring the proxy through the Secret Manager. Once configured, you can deploy the secure API pProxy to the target runtime, such as CloudHub, Runtime Fabric, and on-premises.

Reviewer: pProxy -> proxy

Author: Done! Thanks!

> The certificate is uploaded to a secure vault within API Platform, and then made available when configuring the proxy through the Secret Manager. Once configured, you can deploy the secure API proxy to the target runtime, such as CloudHub, Runtime Fabric, and on-premises.

Reviewer 2: Can this be reworded to use active rather than passive voice?
I'm not sure what our audience does or what is automatic, so I'm limited in what I can suggest here, but here's a guess:

```suggestion
Upload your certificate to a secure vault within API Platform, make it available when you configure the proxy through the secrets manager, and then deploy the secure API proxy to the target CloudHub, Runtime Fabric, or on-premises target.
```

Author: Done! Thanks!

> * xref:proxy-domain-support.adoc[HTTPs defined in already deployed Mule Domain]

Reviewer: "HTTPs defined in already deployed Mule Domain" -> "Certificate defined in already deployed Mule Domain"

Author: Done! Thanks!

> * xref:proxy-domain-support.adoc[Certificate defined in already deployed Mule Domain]

Reviewer 2: ```suggestion
* xref:proxy-domain-support.adoc[Use a certificate defined in an already-deployed Mule domain.]
```

Author: Done! Thanks!

+

> The certificate is configured in the xref:mule-runtime::shared-resources.adoc[domain] and then the API proxy references it. Once configured, you can deploy the secure API pProxy to the target runtime, such as CloudHub, Runtime Fabric, and on-premises. CloudHub is not supported with this approach because,because as it does not support Mule domains.

Reviewer: "The certificate is configured in the xref:mule-runtime::shared-resources.adoc[domain] and then the API proxy references it." -> "The certificate is configured in the domain and then the API proxy HTTPS listener references it."

Author: Done! Thanks!

> The certificate is configured in the domain and then the API proxy HTTPS listener references it. Once configured, you can deploy the secure API pProxy to the target runtime, such as CloudHub, Runtime Fabric, and on-premises. CloudHub is not supported with this approach because,because as it does not support Mule domains.

Reviewer 2: ditto previous edit
```suggestion
Configure the certificate in the domain so that the API proxy HTTPS Listener references it, and then deploy the secure API proxy to the target Runtime Fabric, or on-premises target. (CloudHub is not supported with this approach because it does not support Mule domains.)
```

Author: Done! Thanks!

> * HTTPs configured using Anypoint Studio

Reviewer: "HTTPs configured using Anypoint Studio" -> "Certificate configured using Anypoint Studio"

Author: Done! Thanks!

> * Certificate configured using Anypoint Studio

Reviewer 2: ```suggestion
* Configure your certificate using Anypoint Studio.
```

Author: changed to "Use a certificate configured using Anypoint Studio" to make all bullet points parallel. Done! Thanks!

+

> For this approach, download the API proxy, open it in Anypoint Studio, and xref:mule-runtime::build-an-https-service.adoc[manually configure HTTPs]. You can, then deploy to the proxy to the target runtime, such as CloudHub, Runtime Fabric, and on-premises.  

Reviewer 2: ```suggestion
For this approach, download the API proxy, open it in Anypoint Studio, and xref:mule-runtime::build-an-https-service.adoc[manually configure HTTPS]. You can then deploy the proxy to the target CloudHub, Runtime Fabric, or on-premises target. (If the target runtime is on CloudHub, use port 8082 when working with HTTPs.)
```

Author: Done! Thanks!

> If the target runtime is on CloudHub, use port 8082 when working with HTTPs.

Reviewer 2: I'd move this up to the previous para

Author: Done! Thanks!
